Interest rate on savings feature

sunyamohammed opened the pull request:
Changes Made in this PR:

I have added the functionality of rewarding interest on the customer's balance for every 5 deposits they make into their account, given that the deposit is over $20 and they have no outstanding overdraft balance. This will allow us to attract more customers to Testudo Bank and compete with other banks by incentivizing customers to make deposits to Testudo Bank.  

---

`src/main/java/net/codejava/MvcController.java` 

- Added new constant `MIN_DEPOSIT_FOR_INTEREST` which sets the minimum deposit necessary to apply interest to $20
- Implemented `applyInterest()` which increments the `numDepositsForInterest` column in the `Customers` table for the user if the deposit amount is at least above the minimum and applies the interest if `numDepositsForInterest` is a multiple of 5
- Updates the TransactionHistory to show the added interest as a deposit if the interest was rewarded

---

`src/test/java/net/testudobank/tests/MvcControllerIntegTest.java`

- Wrote test that checks that interest is being applied every 5 transactions and that interest is not applied after the 5th transaction
- Wrote test that tests the edge cases for deposits of $19.99 and $20.01

s1ddh4rthc commented:
Hey Sunya, nice job on your PR comment and your commit messages. Your logic looks good - a recommendation I have is to make your logic reset the `numDepositsForInterest` variable to 0 after an interest deposit, rather than checking if the current numDeposits % 5 == 0.

sunyamohammed commented:
Thanks for the feedback! I've changed my logic in `MvcController.java` to reset `numDepositsForInterest` to 0 after five deposits, and updated my integration tests in `MvcControllerIntegTest.java` to reflect this change.
